### PR TITLE
updates some definitions of `bn.js`

### PR DESCRIPTION
**Pull request by danwbyrne**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/indutny/bn.js/blob/e69c617b3297b99aca429f30842e27979ef9beb5/lib/bn.js#L66>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I change a few things around that were breaking on our end from these new types, also added the static min/max functions that were left out (linked above).

We were getting some errors on our constructors due to the optional arguments not playing around, so I've just added overloads for it *as well as* splitting the `toArrayLike` function so that it has a clearer output type (and the input of ArrayType needed to be `typeof Buffer`). I'm attaching pictures of the errors we were getting previously.

Constructor Error

To Array Like

### CODE REVIEW

File: types/bn.js/index.d.ts [hunk 1]
```diff
@@ -62,7 +66,17 @@ declare class BN {
     /**
      * @description returns true if the supplied object is a BN.js instance
      */
-    static isBN(b: any): b is BN;
```
Comment (RyanCavanaugh) [inline types/bn.js/index.d.ts]: Why was this changed?

Comment (danwbyrne) [reply types/bn.js/index.d.ts]: I had a misunderstanding about the `b is BN` type. I've restored it.
